dpc-portal/app/services/ao_verification_service.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+# A service that verifies a user as an Authorized Official (AO) for a given organization
+class AoVerificationService
+  def initialize
+    @cpi_api_gw_client = CpiApiGatewayClient.new
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/MethodLength
+  # rubocop:disable Metrics/PerceivedComplexity
+  def check_ao_eligibility(organization_npi, hashed_ao_ssn)
+    begin
+      approved_enrollments = get_approved_enrollments(organization_npi)
+      if approved_enrollments == 'bad_npi'
+        Rails.logger.warn "Unable to find organization NPI #{organization_npi}"
+        return { success: false, reason: approved_enrollments }
```

Comment:
nit: should this be more explicitly named? this will make the upstream code clearer too (`result.failure_reason` instead of `result.reason`)

```suggestion
        return { success: false, failure_reason: approved_enrollments }
```